### ISSUE TITLE
Input focus incorrect after mouse selection of snippets 

### ISSUE THREAD

**Issue by kmcnaught** [microsoft/vscode#162729]
Does this issue occur when all extensions are disabled?: **Yes**

- VS Code Version: 
- Version: 1.71.2 (user setup)
Commit: 74b1f979648cc44d385a2286793c226e611f59e7
Date: 2022-09-14T21:03:37.738Z
Electron: 19.0.12
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Windows_NT x64 10.0.19044
Sandboxed: No
- OS Version: Windows 10

Steps to Reproduce:

Use a snippet that has repeat placeholder variables, e.g. 
```
	"If instanceof statement then cast": {
		"prefix": [
			"ific",
			"snippet_if_instanceof_then_cast"
		],
		"body": [
			"if (${1:varName} instanceof ${2:$TM_FILENAME_BASE}) {",
			"\t$TM_SELECTED_TEXT",
			"\t$2 ${3:varName2} = ($2) $1;",
			"\t$0",
			"}"
		],
		"description": "An if instanceof statement then cast"
	},
```

1. Open command palette, select "Insert snippet"
2. In the list of available snippets, select `snippet_if_instanceof_then_cast` **using the mouse to left click on it**
3. Try typing to fill in the highlighted/selected placeholder variable
4. The text editor does not have focus. 
5. (If you manually select the text editor for focus, it is very easy to lose the multiple cursor focus which is essential for efficiently 'filling in' the snippet)

This is an accessibility issue, which arose while using eye control for coding, but the issue persists when just using conventional mouse control.

This video shows the correct behaviour, with focus immediately moved to the text file with correct cursor selection when keyboard control is used to select a snippet. Note that NumPad2 is being used as a shortcut for 'tabbing' to the next placeholder, to avoid other conflicts:
https://www.loom.com/share/b1f209d89ca24e0597c41de4887366c5

This video shows the bug, with focus lost when mouse control is used to select a snippet.
https://www.loom.com/share/faf7848bcf974e6cb5a553aecb1801a8

**Comment by jrieken** [comment]
@TylerLeonhardt I will push a change to explicitly set focus back to the editor after having interacted with quick pick. However, this looks like an issue with quick pick itself, that fact that clicking vs key-down is different seems 🐟y and other might be affected by this as well

**Comment by kmcnaught** [comment]
Thanks for the quick investigation. We are getting on really well with eye control otherwise, and getting snippets working flawlessly is going to be a big win!

**Comment by rebornix** [comment]
@TylerLeonhardt since you are out, I'll move this to next iteration.

**Comment by TylerLeonhardt** [comment]
@jrieken fixed the actual bug... and the focus stuff is covered by https://github.com/microsoft/vscode/issues/158576

**Comment by jrieken** [comment]
Cough, cough. IMO https://github.com/microsoft/vscode/issues/158576 is the actual bug and affected folks are implementing workarounds

**Comment by TylerLeonhardt** [comment]
Agreed
